StyleCop.Analyzers/StyleCop.Analyzers/Lightup/RecordDeclarationSyntaxWrapper.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Lightup;
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+internal partial struct RecordDeclarationSyntaxWrapper : ISyntaxWrapper<TypeDeclarationSyntax>
+{
+    public static implicit operator RecordDeclarationSyntaxWrapper(TypeDeclarationSyntax node)
```

Comment:
This corresponds to a conversion from a base type to a sub type. Adding an implicit conversion operator for that is not a good thing in my book.

Reply:
I see your point, I am happy to make it an explicit operator. With the way it is being used I could also do something like:

```cs
bool RecordDeclarationSyntaxWrapper.TryWrap(TypeDeclarationSyntax node, out RecordDeclarationSyntaxWrapper result)
```

similar to `int.TryParse` and friends. I'm not sure if that's a bit out there though

Reply:
Just to be clear: The explicit conversion operator already exists, so you can just remove this file and add a cast where you get an error.

Reply:
Ah, understood. Removed this file and used the explicit operator.